### ISSUE TITLE
Port for Librem phone/Pinephone?

### ISSUE THREAD

**Issue by programmin1** [streetcomplete/StreetComplete#2756]
**Use case**
I got a Librem phone which doesn't seem to have an equivalent app to this.

**Proposed Solution**
Can this app be ported to a standard Linux app so Linux desktop (and Linux mobile) can use it? 
I wrote about how a standard Linux application can be adjusted to work on the smaller device screens, here:
https://howtotrainyourrobot.com/building-an-amateur-radio-app-for-librem-phone-part-3/

**Comment by HolgerJeromin** [comment]
Some stuff from #1892 applies here, too.

**Comment by matkoniecz** [comment]
I guess that running it in an emulator of some kind is not a viable solution? https://puri.sm/products/librem-5/ has just 3GB RAM so likely not.

> Can this app be ported to a standard Linux app so Linux desktop (and Linux mobile) can use it?

It can be done, it "only" needs someone to do this.

I am not planning to do this. Except quite unlikely case of someone willing to fund months of work on that. And device for testing.

Not checked is there a viable method to run Kotlin based GUI on Linux - is it necessary to write it from scratch or at least use code. Though maybe some parts from #1892 can be skipped - is should be not necessary to rewrite Java parts.

But all GUI code likely needs to be basically redone from scratch (or maybe not? not investigated is there some support for making it easier).

**Comment by westnordost** [comment]
Duplicate of #1386

**Comment by westnordost** [comment]
The librem itself looks like a neat phone (hardware), just looked at a few articles. And stumbled upon this:

https://www.teltarif.de/librem-5-android-linux-apps-emulation-anbox/news/82029.html

<img src="https://www.teltarif.de/img/arch/2020/kw39/librem-5-android-linux-apps-emulation-anbox-2f.jpg"/>

Wat. I guess the author of that article is a fan of StreetComplete 🥰

**Comment by smichel17** [comment]
I am tentatively planning on getting a Librem 5 when my current phone kicks the bucket. I've been thinking of porting [SimpleTask](https://github.com/mpcjanssen/simpletask-android) when that happens. So a few months ago I spent around an hour looking into writing kotlin/native + GTK. I wasn't particularly thrilled with what I found — GTK's C bindings are, well, C-like, and the kotlin interop looked messy to me. Might be able to wrap it to be not so bad, didn't spend the time to figure that out. The other option for native is QT, but then you don't get [libhandy](https://gitlab.gnome.org/GNOME/libhandy/); not sure what QT is like for writing mobile-friendly interfaces.

The other option is using kotlin's js target and porting to a web app instead, which could be packaged for desktop with electron or similar. For SimpleTask I think this makes sense, but StreetComplete is a heavier app so I would guess it would benefit more from being fully native… but maybe I'm wrong.

A middle ground would be staying on the JVM and using JavaFX I guess? Also not sure about that + mobile.
